tensorflow_probability/python/sts/components/autoregressive_moving_average.py

```diff
@@ -0,0 +1,238 @@
+
+    This is formulated as a state space model by letting the latent state encode
+    the most recent values; see 'Mathematical Details' below.
+
+    The parameters `level_scale` and `observation_noise_scale` are each (a batch
+    of) scalars, and `coefficients` is a (batch) vector of size `[order]`. The
+    batch shape of this `Distribution` is the broadcast batch
+    shape of these parameters and of the `initial_state_prior`.
+
+    #### Mathematical Details
+
+    The Hamilton autoregressive moving average model implements a
+    `tfp.distributions.LinearGaussianStateSpaceModel` with `latent_size = order`
+    and `observation_size = 1`. The latent state vector encodes the recent history
+    of the process, with the current value in the topmost dimension. At each
+    timestep, the transition sums the previous values to produce the new expected
+    value, shifts all other values down by a dimension, and adds noise to the
+    current value. This is formally encoded by the transition model:
+
+    ```
+    transition_matrix = [ phi[0], phi[1],     ..., phi[p]
+                          1.,       0 ,       ..., 0.
+                          0.,       1.,       ..., 0.
+                          ...
+                          0.,       0.,  ...,  1.,  0.   ]
+
+    transition_noise ~ N(loc=0., scale=diag([level_scale, 0., 0., ..., 0.]))
+    ```
+
+    The observation model simply extracts the current (topmost) value,
+    sums the previous noise and optionally adds independent noise at each step:
+
+    ```
+    observation_matrix = [1, theta[0], theta[1], ..., theta[p-1]]
+    observation_noise ~ N(loc=0, scale=observation_noise_scale)
+    ```
+
+    Models with `observation_noise_scale = 0.` are ARMA(p, p-1) processes
+    in the formal sense. Setting `observation_noise_scale` to a nonzero value
+    corresponds to a latent ARMA(p, p-1) process observed under an iid noise model.
+
+    See the [Wikipedia article](http://web.pdx.edu/~crkl/readings/Hamilton94.pdf)
+    for details on the Hamilton state space formulation for ARMA(p, p-1) processes.
+    """
+
+    def __init__(self,
+                 num_timesteps,
+                 ar_coefficients,
+                 ma_coefficients,
+                 level_scale,
+                 initial_state_prior,
+                 observation_noise_scale=0.,
+                 name=None,
+                 **linear_gaussian_ssm_kwargs):
+        """ Build a state space model implementing an ARMA(p, p - 1) process.
+
+        Args:
+            num_timesteps: Scalar `int` `Tensor` number of timesteps to model
+                with this distribution.
+            ar_coefficients: `float` `Tensor` of shape `concat(batch_shape, [order])`
+                defining  the autoregressive coefficients. The coefficients are defined
+                backwards in time: `coefficients[0] * level[t] + coefficients[1] *
+                level[t-1] + ... + coefficients[order-1] * level[t-order+1]`.
```

Comment:
I actually had a mistake on `ma_coefficients` as the current noise `noise[t]` does not get a `ma_coefficients`. So it should read:
```
ma_coefficients: `float` `Tensor` of shape `concat(batch_shape, [order])`
    defining  the moving average coefficients. The ma_coefficients are defined
    backwards in time: `noise[t] + ma_coefficients[0] * noise[t-1]
    + ... + ma_coefficients[order-2] * noise[t-order+1]`.
```
But otherwise cleaned the names up in the docs.